### PR TITLE
Allow items with highest priority to be pushed on queue when full

### PR DESCRIPTION
**Pull request by jpbruinsslot**
### Changes

Allow items with highest priority to be pushed on queue when full

### Issue link

Closes https://github.com/minvws/nl-kat-coordination/issues/1889

### CODE REVIEW

File: mula/tests/integration/test_api.py [hunk 1]
```diff
@@ -151,19 +151,37 @@ def test_push_queue_full(self):
...
         self.assertEqual(1, self.scheduler.queue.qsize())
 
+    def test_push_queue_full_high_priority(self):
+        # Set maxsize of the queue to 1
+        self.scheduler.queue.maxsize = 1
+
+        # Add one task to the queue
+        first_item = create_p_item(self.organisation.id, 1)
+        response = self.client.post(
+            f"/queues/{self.scheduler.scheduler_id}/push", json=json.loads(first_item.model_dump_json())
```
Comment (ammar92) [inline mula/tests/integration/test_api.py:174]: The serialization and deserialization process doesn't make sense, either:
- `json=first_item.model_dump()` or
- `data=first_item.model_dump_json()`

File: mula/tests/integration/test_api.py [hunk 2]
```diff
@@ -151,19 +151,37 @@ def test_push_queue_full(self):
...
+        )
+        self.assertEqual(response.status_code, 201)
+        self.assertEqual(1, self.scheduler.queue.qsize())
+
+        # Try to add another task to the queue through the api
+        second_item = create_p_item(self.organisation.id, 1)
+        response = self.client.post(f"/queues/{self.scheduler.scheduler_id}/push", json=json.loads(second_item.json()))
```
Comment (ammar92) [inline mula/tests/integration/test_api.py:181]: See previous comment

**Review by ammar92 — COMMENTED** [review]
Tiny remark on the tests; looks good otherwise 👍

**Review by ammar92 — APPROVED** [review]
No more remarks

**Comment by Darwinkel** [comment]
## Checklist for QA:
- [x] I have checked out this branch, and successfully ran a fresh `make reset`.
- [x] I confirmed that there are no unintended functional regressions in this branch:
    - [x] I have managed to pass the onboarding flow
    - [x] Objects and Findings are created properly
    - [x] Tasks are created and completed properly
- [x] I confirmed that the PR's advertised `feature` or `hotfix` works as intended.

### What works:
- There is a `queue is full` TooManyRequestsError error on `main` when using `SCHEDULER_PQ_MAXSIZE=1`
- There is no `queue is full` TooManyRequestsError error on this branch when using `SCHEDULER_PQ_MAXSIZE=1`
- No apparent regressions

**Review by underdarknl — APPROVED** [review]

**Pull request merged**